core/src/main/scala-3/chisel3/Module.scala

```diff
@@ -0,0 +1,29 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chisel3
+
+import chisel3.experimental.{BaseModule, SourceInfo}
+
+object Module extends ObjectModuleImpl with SourceInfoDoc {
+
+  /** A wrapper method that all Module instantiations must be wrapped in
+    * (necessary to help Chisel track internal state).
+    *
+    * @param bc the Module being created
+    *
+    * @return the input module `m` with Chisel metadata properly set
+    */
+  // TODO(adkian-sifive) the callsite here explicitly passes
+  // sourceInfo so it cannot be a contextual parameter
```

Comment:
This comment looks to be no longer true? But even if it were, you can always `summon[_]`.

Reply:
so the comment here is referring to this callsite: https://github.com/chipsalliance/chisel/blob/dc39308e583a793c08dd124a42b5c5b8a0a1cf53/core/src/main/scala/chisel3/internal/package.scala#L133

i think we need the do_apply here? (same comment for below)

Reply:
That callsight should call `_applyImpl`, that's the replacement for `do_apply`. This is the same thing as the other places where we got rid of `do_` versions of things because we no longer need the macro on apply, we can just take a contextual parameter.